src/context/WalletProvider/MetaMask/components/Connect.tsx

```diff
@@ -34,36 +35,66 @@ export interface MetaMaskSetupProps
 }
 
 export const MetaMaskConnect = ({ history }: MetaMaskSetupProps) => {
+  const translate = useTranslate()
   const isMetaMaskMobileWebView = checkIsMetaMaskMobileWebView()
-  const { dispatch, getAdapter, onProviderChange } = useWallet()
+  const {
+    dispatch,
+    getAdapter,
+    state: { modalType },
+  } = useWallet()
   const localWallet = useLocalWallet()
   const [loading, setLoading] = useState(false)
   const [error, setError] = useState<string | null>(null)
   const showSnapModal = useSelector(selectShowSnapsModal)
 
+  const detectedMipdProviders = useSyncExternalStore(mipdStore.subscribe, mipdStore.getProviders)
+  const mipdProviders = useMemo(
+    () => uniqBy(detectedMipdProviders.concat(staticMipdProviders), 'info.rdns'),
+    [detectedMipdProviders],
+  )
+  const maybeMipdProvider = mipdProviders.find(provider => provider.info.rdns === modalType)
+
+  const headerText: [string, InterpolationOptions] = useMemo(
+    () => [
+      'walletProvider.mipd.connect.header',
+      { name: maybeMipdProvider?.info.name ?? 'MetaMask' },
+    ],
+    [maybeMipdProvider?.info.name],
+  )
+
+  const bodyText: [string, InterpolationOptions] = useMemo(
+    () => [
+      'walletProvider.mipd.connect.body',
+      { name: maybeMipdProvider?.info.name ?? 'MetaMask' },
+    ],
+    [maybeMipdProvider?.info.name],
+  )
```

Comment:
suggestion: as said previously, I think we could manage those MIPD informations from the WalletProvider as an abstraction might be risky though

Reply:
See https://github.com/shapeshift/web/pull/7869#discussion_r1795796004
For `<Connect />` this is particularly obvious why we may not want this: we don't have a wallet just yet, so mipd data wouldn't work in the modal and we would either throw or have wrong rdns info (MM icon and name mostly) since there isn't a connected wallet yet